guides/source/error_reporting.md

```diff
@@ -0,0 +1,151 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
+
+Error Reporting
+========================
+
+This guide introduces ways to manage exceptions occur in Ruby on Rails applications.
+
+After reading this guide, you will know:
+
+* How to use Rails' error reporter to capture and report errors.
+
+--------------------------------------------------------------------------------
+
+Error Reporter
+------------------------
+
+The [error reporter](https://api.rubyonrails.org/classes/ActiveSupport/ErrorReporter.html) collects exceptions occur in Ruby on Rails applications and reports them to registered subscribers.
+
+The goal is to replace such manual rescue in applications and libraries
+
+```rb
+begin
+  do_something
+rescue SomethingIsBroken => error
+  MyErrorReportingService.notify(error)
+end
+```
+
+with
+
+```rb
+Rails.error.handle do
+  do_something
+end
+```
+
+This approach provides several benefits:
+
+* It decouples application code from error reporting libraries.
+* It reduces boilterplate code for handling exceptions.
+* Error reporting libraries will need less monkey-patches and be less intrusive to applications.
```

Comment:
I'd add that it allows Rails aware or at least Active Support aware libraries, to report handled errors to the host application. That was the main argument for introducing this interface.

Reply:
👍 added as the first benefit